scripts/test_ekf.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+import rospy
+import numpy as np
+import matplotlib.pyplot as plt
+from geometry_msgs.msg import PoseWithCovarianceStamped
+from nav_msgs.msg import Odometry
+from mrover.msg import Waypoint, WaypointType
+from smach_msgs.msg import SmachContainerStatus
+import message_filters
+from util.course_publish_helpers import convert_waypoint_to_gps, publish_waypoints
+from util.SE3 import SE3
+
+
+class EKF_Test:
+    def __init__(self):
+        self.raw_data = []
+        self.ekf_data = []
+        self.gt_data = []
+        self.timestamps = []
+        self.nav_state = ""
+
+        # subscribe to both odometry topics and synchronize them
+        raw_sub = message_filters.Subscriber("gps/pose", PoseWithCovarianceStamped)
+        ekf_sub = message_filters.Subscriber("global_ekf/odometry", Odometry)
+        truth_sub = message_filters.Subscriber("ground_truth", Odometry)
+        rospy.Subscriber("smach/container_status", SmachContainerStatus, self.nav_status_callback)
+
+        ts = message_filters.TimeSynchronizer([raw_sub, ekf_sub, truth_sub], 10)
+        ts.registerCallback(self.odoms_callback)
+
+    def odoms_callback(self, raw_pose_msg: PoseWithCovarianceStamped, ekf_odom_msg: Odometry, truth_odom_msg: Odometry):
+        msgs = [raw_pose_msg, ekf_odom_msg, truth_odom_msg]
+        datas = [self.raw_data, self.ekf_data, self.gt_data]
+
+        self.timestamps.append(raw_pose_msg.header.stamp.to_sec())
+        for msg, data in zip(msgs, datas):
+            pos = msg.pose.pose.position
+            q = msg.pose.pose.orientation
+            point = np.array([pos.x, pos.y, pos.z, q.x, q.y, q.z, q.w])
+            data.append(point)
+
+    def nav_status_callback(self, status_msg):
+        self.nav_state = status_msg.active_states[0]
+
+    def execute_path(self):
+        square_path = [
```

Comment:
`square_path` is unused in this function, should be removed or if it's intended in a different context, should be in moved

Reply:
Actually, if you're making that and the `line_path` var in this class, might want to instantiate them as constants at the top of the class if they aren't being modified/are constant

Reply:
yeah these were just some test paths I was playing with, I'll add them as constants